client/src/lib/components/CaptchaV2.svelte

```diff
@@ -22,21 +23,43 @@
           : theme;
       const mobileScreen = window.innerHeight > window.innerWidth;
 
-      if (!window.grecaptcha) {
+      if (captchaProvider === CaptchaProvider.procaptcha) {
+        if (!window.procaptcha) {
+          captchaError = true;
+          throw new Error(`${captchaProvider} is undefined!`);
+        }
+        window.procaptcha.render(captchaId, {
+          siteKey: captchaKey,
+          theme: colorTheme,
+          callback: "onToken",
+          "chalexpired-callback": "onExpiredToken",
+          captchaType: "image",
+        });
+      } else if (captchaProvider === CaptchaProvider.recaptcha) {
+        if (!window.grecaptcha) {
+          captchaError = true;
+          throw new Error("grecaptcha is undefined!");
+        }
+        window.grecaptcha.render(captchaId, {
+          sitekey: captchaKey,
+          theme: colorTheme,
+          callback: "onToken",
+          size: mobileScreen ? "compact" : "normal",
+          "expired-callback": "onExpiredToken",
+        });
+      } else {
         captchaError = true;
-        throw new Error("grecaptcha is undefined!");
+        throw new Error(`Unknown captcha provider: ${captchaProvider}`);
       }
-      window.grecaptcha.render(captchaId, {
-        sitekey: captchaKey,
-        theme: colorTheme,
-        callback: "onToken",
-        size: mobileScreen ? "compact" : "normal",
-        "expired-callback": "onExpiredToken",
-      });
     };
 
     window.onToken = (token) => {
       dispatch("token", token);
+      // Forces a new captcha on page reload
+      if (captchaProvider === CaptchaProvider.procaptcha) {
+        window.localStorage.removeItem("@prosopo/current_account");
+        window.localStorage.removeItem("@prosopo/provider");
```

Comment:
Out of curiosity, why are these elements removed?

Reply:
My bad, these items should have been updated to read

```
window.localStorage.removeItem("@prosopo/procaptcha");
```

We have a mechanism in place that stores the following in a hex string in local storage:

```
export const ProsopoLocalStorageSchema = z.object({
    account: z.string().optional(),
    blockNumber: z.number().optional(),
    providerUrl: z.string().optional(),
})
```

If all 3 are present, it means the user has completed a captcha before from `providerUrl`. When the user returns to the page, we contact the provider and ask if the user was previously verified. If they were, we don't force them to do a captcha again. *However*, in your reCAPTCHA implementation the user was forced to complete a captcha on each page load. We mimic this behaviour by removing the local storage.

It's worth noting that the default verified time is set to 60s. So even if there was local storage, it would only be valid for 60s. Happy to change the way this works if you like?